hokusai/services/yaml_spec.py

```diff
@@ -1,21 +1,22 @@
 import os
+import atexit
 
 import yaml
 
-from tempfile import NamedTemporaryFile
-
-from hokusai.lib.config import config
+from hokusai.lib.config import config, HOKUSAI_TMP_DIR
 from hokusai.lib.config_loader import ConfigLoader
 from hokusai.lib.template_renderer import TemplateRenderer
 
 from hokusai.lib.exceptions import HokusaiError
 
 from hokusai.services.ecr import ECR
 
-class KubernetesSpec(object):
+class YamlSpec(object):
   def __init__(self, kubernetes_yaml):
     self.kubernetes_yaml = kubernetes_yaml
     self.ecr = ECR()
+    self.tmp_filename = None
+    atexit.register(self.cleanup)
```

Comment:
Magic here to clean up tmp file handles when the interpreter exits... @dleve123 you mentioned wanting to take a look at the rendered Yaml files - we could add a flag to leave them in place?

Reply:
oh yeah, do think it would be nice to see the final rendered template in some debug mode

Reply:
Cool so now if `DEBUG` env var is set hokusai will leave the rendered templates in `.hokusai-tmp` rather than printing to stderr

Reply:
See https://github.com/artsy/hokusai/pull/199 for a couple of tweaks for template debugging